test/common/upstream/cluster_manager_impl_test.cc

```diff
@@ -164,6 +167,167 @@ envoy::config::bootstrap::v3::Bootstrap defaultConfig() {
   return parseBootstrapFromV3Yaml(yaml);
 }
 
+class ODCDTest : public ClusterManagerImplTest {
+public:
+  void SetUp() override {
+    create(defaultConfig());
+    odcds_ = MockOdCdsApi::create();
+    odcds_handle_ = cluster_manager_->createOdCdsApiHandle(odcds_);
+  }
+
+  void TearDown() override {
+    odcds_.reset();
+    odcds_handle_.reset();
+    factory_.tls_.shutdownThread();
+  }
+
+  ClusterDiscoveryCallbackSharedPtr createCallback() {
+    return std::make_shared<ClusterDiscoveryCallback>(
+        [this](ClusterDiscoveryStatus cluster_status) {
+          UNREFERENCED_PARAMETER(cluster_status);
+          testFunction();
+        });
+  }
+
+  ClusterDiscoveryCallbackSharedPtr createCallback(ClusterDiscoveryStatus expected_cluster_status) {
+    return std::make_shared<ClusterDiscoveryCallback>(
+        [this, expected_cluster_status](ClusterDiscoveryStatus cluster_status) {
+          EXPECT_EQ(expected_cluster_status, cluster_status);
+          testFunction();
+        });
+  }
+
+  MOCK_METHOD(void, testFunction, ());
+
+  MockOdCdsApiSharedPtr odcds_;
+  OdCdsApiHandlePtr odcds_handle_;
+  std::chrono::milliseconds timeout_ = std::chrono::milliseconds(5000);
+};
+
+TEST_F(ODCDTest, TestAllocate) {
+  envoy::config::core::v3::ConfigSource config;
+  OptRef<xds::core::v3::ResourceLocator> locator;
+  ProtobufMessage::MockValidationVisitor mock_visitor;
+
+  config.mutable_api_config_source()->set_api_type(
+      envoy::config::core::v3::ApiConfigSource::DELTA_GRPC);
+  config.mutable_api_config_source()->set_transport_api_version(envoy::config::core::v3::V3);
+  config.mutable_api_config_source()->mutable_refresh_delay()->set_seconds(1);
+  config.mutable_api_config_source()->add_grpc_services()->mutable_envoy_grpc()->set_cluster_name(
+      "static_cluster");
+
+  auto handle = cluster_manager_->allocateOdCdsApi(config, locator, mock_visitor);
+  EXPECT_NE(handle, nullptr);
```

Comment:
What is this test verifying? I don't think it's possible for `handle` to be a `nullptr`...

Reply:
Frankly speaking it's about improving the coverage, because I could not use `allocateOdCdsApi` in other tests as I had no way to inject a mock odcds there.